server/game_service.py

```diff
@@ -105,24 +104,24 @@ async def update_data(self):
             # Turn resultset into a list of uids
             self.ranked_mods = set(map(lambda x: x[0], rows))
 
-    @property
-    def dirty_games(self):
-        return self._dirty_games
-
-    @property
-    def dirty_queues(self):
-        return self._dirty_queues
-
     def mark_dirty(self, obj: Union[Game, MatchmakerQueue]):
         if isinstance(obj, Game):
             self._dirty_games.add(obj)
         elif isinstance(obj, MatchmakerQueue):
             self._dirty_queues.add(obj)
 
-    def clear_dirty(self):
+    def get_new_dirty_games(self) -> Set[Game]:
```

Comment:
`get` here doesn't necessarily communicate the side effect of resetting `self._dirty_games`, which might end up surprising someone? I'm not sure what would be better, `dump`, `pop`, `get_and_reset_dirty_*`?

Reply:
I wasn't really sure either. That's why I called it `get_new` instead of just `get`. Not really a fan of `get_and` though. Maybe `pop` is better.